File: src/Fertilizer.java
public class Fertilizer {
    public final String name;
    public final int duration;
    public final int cost;
    public final String effect;

    public Fertilizer(String name, int duration, int cost, String effect) {
        this.name = name;
        this.duration = duration;
        this.cost = cost;
        this.effect = effect;
    }

    public boolean affectsType(String cropType) {
        switch (effect) {
            case "BoostPlant": return cropType.equals("Plant");
            case "BoostMushroom": return cropType.equals("Mushroom");
            case "BoostInsect": return cropType.equals("Insect");
            default: return false;  // Catalyst etc.
        }
    }

    @Override
    public String toString() {
        return name;
    }
}

File: src/Fertilizer.py
# Fertilizer.java

class Fertilizer:
    def __init__(self, name, duration, cost, effect):
        self.name = name
        self.duration = duration
        self.cost = cost
        self.effect = effect

    def affects_type(self, crop_type):
        if self.effect == "BoostPlant":
            return crop_type == "Plant"
        elif self.effect == "BoostMushroom":
            return crop_type == "Mushroom"
        elif self.effect == "BoostInsect":
            return crop_type == "Insect"
        else:
            return False  # For effects like Catalyst, etc.

    def __str__(self):
        return self.name

